### DISCUSSION TITLE
Offene Forschungsfragen / Open Scientific Questions

### DISCUSSION THREAD

**Discussion by christianheimann** [e-id-admin/governance-sounding-board#45] [Ideas]
Immer wieder wird das E-ID-Team angefragt, welche Fragestellungen z.B. durch eine Master-Thesis geklärt werden könnten. Ebenso werden konkrete Fragestellungen ans Team herangetragen, die eine wissenschaftliche Aufarbeitung benötigen. Dieser Thread soll offene Forschungsfragen auflisten, Denk- und Antwortansätze bekannt machen, der Vernetzung und Präzisierung der Fragen helfen.
___________

L'équipe E-ID est régulièrement demandée, quelles questions ouvertes existent qui pourront être répondu par exemple dans le cadre d'une thèse de master. Egalement, des questions concrètes nécessitant un traitement scientifique sont soumises à l'équipe. Ce thread  a pour but d'énumérer les questions de recherche en suspens, de faire connaître des pistes de réflexion et de réponse, d'aider à la mise en réseau et à la précision des questions posées.
____________

Al team E-ID viene ripetutamente chiesto quali domande potrebbero essere chiarite, ad esempio, da una tesi di Master. Il team riceve anche domande concrete che richiedono un'analisi scientifica. Questo thread ha lo scopo di elencare le domande di ricerca aperte, di rendere noti gli approcci al pensiero e le risposte, di aiutare il networking e di specificare le domande.
____________

The E-ID team is repeatedly asked which questions could be clarified, for example, by a Master's thesis. The team also receives concrete questions that require scientific analysis. This thread is intended to list open research questions, to make approaches to thinking and answers known, to help networking and to specify the questions.

**Comment by AnnettLaube** [comment]
**VC Signature schemes & Privacy**

Classic or traditional signature schemes only guarantee integrity, authenticity and unforgeability.

But today, privacy requirements are becoming more and more important.  Therefore, the signature schemes should fulfill the following requirements:

1.	_Selective disclosure_: the holder can choose which subset of attribute (inc. the identifier) he/she discloses to a verifier.

2.	_Derived Attributes_ (predicates), e.g., age or liquidity.

3.	_Signature Blinding_: the signature of the issuer (which is a unique value) must be randomized before it is shared with a verifier, to prevent disclosure of a correlatable value.

4.	_Private Holder Binding_:  a credential is bound to a holder without creating a correlating factor for the holder that must be disclosed upon presentation.

Some existing schemes, like BBS+, meet some of the requirements, but do not allow derived attributes.

**Comment by AnnettLaube** [comment]
**Anonymity Preserving Revocation of VCs**

VCs have often a limited validity, but also then, it should be possible to revoke them, e.g., in case of a lost smartphone.  _But revocation and the verification of revoked identities or VCs may not leak additional information._

In the verifiable credential specification of W3C, the problem of revoking a credential is solved by adding a property _credentialStatus_ containing a URL to the credential's status. Using this URL, a verifier can then request the status from this issuer. The downside of this procedure is that every time a VC is used, the issuer is alerted, and he can potentially _track the usage_ of all its issued VCs.

As alternative, a “privacy-preserving mechanism for revoking Verifiable Credentials” was drafted in https://w3c-ccg.github.io/vc-status-rl-2020/. The general idea is to aggregate many credential statuses into one compressed bitstring and publishing that on a HTTP endpoint. Assuming the bitstring is long enough and there are enough verifiable credentials linked to a single bitstring, it is much harder to learn when an individual verifiable credential was used and to whom it was presented. Note that the issuer still can monitor the usage of VCs, if he uses a bitstring for each credential. In addition, the verifier only learns the credential's status but not the reason of the revocation. Depending on the use case this might be advantageous or disadvantageous. 

An other idea is to store the bitstrings on public distributed ledgers…

Beside the privacy-preserving aspects, the approaches differ also in the implementation effort end the possible actuality der revocation information, maybe in more.…

**Reply by mdoujak** [reply]
I have read about the bitstring revocation mechanism and I would like to add, that aside from the reason you also cannot easily determine the point in time the revocation took place. 
In digital signature use cases this is a critical lack of information since I want to verify a signature against the point in time the verification was made and not against now().

The second point is really a question. The way I understood the proposal, the VCs have to be numbered and the number indicates the position in the bitstring. In other words, I use this number from the VC to check the revocation status. If this is correct, two verifiers can use this number as an identifier to correlate the activities of a person across multiple verifiers. I think this is a privacy issue.

**Reply by mdoujak** [reply]
I find the inclusion of a credentialStatus URL a good solution, but I disagree with the implementation. In digital signatures, the OCSP response of the X.509 certificate status check is included in the signature to prove that at the time of the signing, this X.509 certificate was valid.

If we take this approach to the SSI world, we would demand that every time a holder creates a verifiable presentation the credentialStatus is checked and added to the presentation. This would prove that at the time of presentation the VC was valid. Since the holder requests the status from the issuer, privacy is preserved. 

The status response from the issuer should contain the following information:
- status code
- time of revocation
- reason for revocation
- issuer
- hash of the VC this status applies to

What do you think of such an approach?

**Reply by AnnettLaube** [reply]
One of the advantages of VC is the separation of issuing and use of VC: the issuer does not know when a holder is using the VC to create a presentation. This is a privacy feature.  Exactly this advantage is lost,  if the holder requests every time the credentialStatus from the issuer before creating a presentation.

**Reply by mdoujak** [reply]
In my current understanding, I have not yet seen a revocation mechanism that can 100% preserve my privacy. As I understand it I have the following options to loose privacy:

1. I can let the issuer know that I am using the VC issued to me at this time. (bitstring revocation)
2. I can provide the verifier with an identifier that permits the verifier to correlate the VC with other verifiers. (credentialStatus incl. in VP)
3. I can force the verifier to let the issuer know that I want to present the VC to this issuer (credentialStatus queried by verifier)

I think that sharing the time of using the VC is the least privacy invading option. But that is just my opionion.

**Reply by cybee42** [reply]
How about AnonCreds' approach of proofing non-revocation?

https://hyperledger-indy.readthedocs.io/projects/hipe/en/latest/text/0011-cred-revocation/README.html

**Comment by cybee42** [comment]
The Cyber-Defence Campus offers many [student projects](https://cyber-defence-campus.github.io/projects/), some dealing with SSI. So if you are looking for a master thesis or have an intriguing topic that could be studied in a thesis, please talk to us.

Currently available SSI theses:
- [Privacy-preserving and Scalable Revocation of Verifiable Credentials](https://cyber-defence-campus.github.io/projects/projects_topics/topic_2024_SSI_Credential_Revocation/)
- [KERI under scrutiny](https://cyber-defence-campus.github.io/projects/projects_topics/topic_BUMA_2024_KERI/)
- [Evaluation of Self-Sovereign Identity (SSI) Systems](https://cyber-defence-campus.github.io/projects/projects_topics/topic_2024_SSI_Evaluation/)

**Reply by cybee42** [reply]
Follow-up: Two master theses have been completed. You can find the reports and corresponding POC code in these repositories:

- [KERI Under Scrutiny: Use-Case Study for the Swiss e-ID](https://github.com/luffa99/KERI-Under-Scrutiny-Master-Thesis/) 
- [Scalable and Privacy-Preserving Revocation of Verifiable Credentials](https://github.com/alecolo129/eid-revocation-rs)
